docs/transaction-building.md

```diff
@@ -0,0 +1,264 @@
+:::
+
+## Building a transaction
+
+For transaction building, Fleet provides the `TransactionBuilder` and the `OutputBuilder` classes. Both deliver a seamless way to construct transactions with built-in validations, selection strategies, automatic change calculation, _et cetera_.
+
+```ts
+import { TransactionBuilder, OutputBuilder } from "@fleet-sdk/core";
+```
+
+## Step. 1: Instantiating and adding the Creation Height
+
+The `TransactionBuilder` class requires a **Creation Height** as a constructor param.
+
+Transactions end with newly created boxes, or Output boxes. Each newly created Output box needs to contain its creation block height. Usually, the Creation Height is the current block height at the time of the creation of the transaction. Creation height is a required parameter to begin building a transaction.
+
+```ts
+new TransactionBuilder(creationHeight);
+```
+
+## Step. 2: Add inputs
+
+Input boxes are the source of the funds for the transaction.
+
+To add inputs to the transaction body, use the `from()` method. This method can accept a single input box object or an array of input boxes.
+
+<!-- prettier-ignore -->
+```ts
+new TransactionBuilder(creationHeight)
+  .from(inputs);
+```
+
+### Input box object type
+
+Input boxes can be fetched from the blockchain or directly from the [user's wallet](./wallet-interaction.md#step-5-fetch-boxes).
+
+```ts
+type Box = {
+  boxId: string;
+  value: string | bigint;
+  assets: { tokenId: string; amount: string | bigint }[];
+  ergoTree: string;
+  creationHeight: number;
+  additionalRegisters: NonMandatoryRegisters;
+  index: number;
+  transactionId: TransactionId;
+};
+```
+
+## Step. 3: Add Data-Inputs
+
+The Data-Inputs are boxes whose data can be referenced and used by contracts, if required. A good use case for Data-Inputs is using oracle data. For example, the [SigmaUSD](https://sigmausd.io/) contract uses an oracle of Ergo price to set the exchange rate for SigUSD and SigRSV conversation to and from ERG at regular intervals.
+
+- Any **unspent** box can be used as Data-Input.
+- Data-Inputs are **optional** and must be only included in the transaction if required by a contract.
+- Boxes will not be spent by the transaction if used as Data-Input.
+
+If required by a contract, use the `withDataFrom()` method to add Data-Inputs. Similar to the `from()` method, `withDataFrom()` can accept a single input box object or an array of input boxes.
+
+<!-- prettier-ignore -->
+```ts
+new TransactionBuilder(creationHeight)
+  .from(inputs)
+  .withDataFrom(dataInputs);
+```
+
+## Step. 4: Add outputs
+
+Outputs boxes are the destination of funds held by the input boxes. When confirmed by the blockchain, output boxes are ready to be used as input boxes in further transactions.
+
+In the following example, we are using the `OutputBuilder` and the ` TransactionBuilder`'s `to()` method to construct and include outputs in the transaction body.
```

Comment:
```suggestion
In the following example, the `OutputBuilder` and the ` TransactionBuilder`'s `to()` methods are used to construct and include outputs in the transaction body.
```

Reply:
Normally, people suggest avoiding the passive voice, but I think I prefer it here because of the step-by-step of the tutorial. You don't want to put yourself between the user and the actions the document is suggesting.

Reply:
Great! I will try to avoid this in future texts.